tests/test_main.py

```diff
@@ -0,0 +1,162 @@
+#
+# Sonar Scanner Python
+# Copyright (C) 2011-2024 SonarSource SA.
+# mailto:info AT sonarsource DOT com
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+#
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+import unittest
+
+from unittest.mock import patch
+from io import StringIO
+
+from pysonar_scanner.__main__ import initialize_configuration
+from pysonar_scanner.configuration import Configuration, Scanner, Internal, Sonar
+
+
+class TestMain(unittest.TestCase):
+
+    @patch("sys.argv", ["myscript.py"])
+    def test_missing_cli_args(self):
+        with patch("sys.stderr", new=StringIO()) as mock_stderr:
+            with self.assertRaises(SystemExit):
+                initialize_configuration()
```

Comment:
This can also be written as:
```suggestion
        with patch("sys.stderr", new=StringIO()) as mock_stderr, self.assertRaises(SystemExit):
            initialize_configuration()
```